src/memory/AmdFlash.cc

```diff
@@ -676,39 +676,37 @@ bool AmdFlash::checkCommandQuadrupleByteProgram()
 bool AmdFlash::checkCommandBufferProgram()
 {
 	static constexpr std::array<uint8_t, 2> cmdSeq = {0xaa, 0x55};
-	if (chip.program.bufferCommand && partialMatch(cmdSeq)) {
-		if (cmd.size() <= 2) return true;
-		if (cmd.size() >= 3 && cmd[2].value == 0x25) {
-			if (cmd.size() <= 3) return true;
+	if (chip.program.bufferCommand && partialMatch(cmdSeq) && (cmd.size() <= 2 || cmd[2].value == 0x25)) {
+		if (cmd.size() <= 3) {
+			return true;
+		} else if (cmd.size() <= 4) {
+			if (cmd[3].value < chip.program.pageSize && getSector(cmd[3].addr) == getSector(cmd[2].addr)) {
```

Comment:
This patch itself is fine. Though while reading it, i was still thinking about `Sector::operator==()` (introduced in the first patch in this PR).

That operator compares the full struct, and that's correct. But wouldn't it also be correct to only compare the `address` field? My reasoning: no two sectors within the same AmdFlash have the same `address`, so that field by itself is already sufficient to uniquely identify the sector.

And that would also make `operator==` and `operator<=>` 'consistent'. Now that I think about it, why do you need `operator<=>`? It isn't used (I checked). Remove it?

Reply:
Hmm, I was under the assumption that `bool operator==(const Sector& sector) const = default;` used the `operator<=>` implementation, but that’s not the case. I’ll fix it.

I can make use of the `operator<=>` in `isWritable()`.

Reply:
IIRC in the initial proposal for operator<=> it worked like that. operator<=> would be used to derive an operator== (i think even without an explicit `operator==() =default` request). But that was removed from the proposal before it was approved for the c++20 standard.

The reason is performance. Often a dedicated operator== can be (much) faster than using operator<=> to derive operator==. And the fear was that if writing operator== was no longer needed, too many people would omit it (and then get a working but slower version). Many other programming languages (e.g. Rust) also work like this: you need to both implement equality and comparison.

A typical example is a string class. When given two strings with different size, operator== can immediately return false. But operator<=> must loop over the characters until it finds a mismatch.